Snowflake.Data/Core/SFResultSetArrow.cs

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Snowflake.Data.Log;
+using Snowflake.Data.Client;
+using Apache.Arrow;
+using Apache.Arrow.Ipc;
+
+namespace Snowflake.Data.Core
+{
+    class SFResultSetArrow : SFBaseResultSet
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFResultSetArrow>();
+        
+        private int _currentChunkRowIdx;
+        private int _currentChunkRowCount;
+        private readonly int _totalChunkCount;
+        private IResultChunk _currentChunk;
+        private readonly IChunkDownloader _chunkDownloader;
+
+        public SFResultSetArrow(QueryExecResponseData responseData, SFStatement sfStatement, CancellationToken cancellationToken) : base()
+        {
+            try
+            {
+                columnCount = responseData.rowType.Count;
+                _currentChunkRowIdx = -1;
+
+                using (var stream = new MemoryStream(Convert.FromBase64String(responseData.rowsetBase64)))
+                {
+                    using (var reader = new ArrowStreamReader(stream))
+                    {
+                        var recordBatch = reader.ReadNextRecordBatch();
+                        _currentChunkRowCount = recordBatch.Length;
+                        _currentChunk = new SFArrowResultChunk(recordBatch);
+                    }
+                }
+
+                this.sfStatement = sfStatement;
+                updateSessionStatus(responseData);
+
+                if (responseData.chunks != null)
+                {
+                    // TODO - support for multiple chunks
+                    throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+                }
+
+                responseData.rowSet = null;
+
+                sfResultSetMetaData = new SFResultSetMetaData(responseData);
+
+                isClosed = false;
+
+                queryId = responseData.queryId;
+            }
+            catch(System.Exception ex)
+            {
+                s_logger.Error("Result set error queryId="+responseData.queryId, ex);
+                throw;
+            }
+        }
+
+        internal override async Task<bool> NextAsync()
+        {
+            if (isClosed)
+            {
+                throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);
+            }
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_chunkDownloader != null)
```

Comment:
`!=` or `==`?

Reply:
`!=` but changed the condition to be more clear